### PR TITLE
[CatalogPromotions] Fixed percentage discount prices rounding

### LINKED ISSUE #14522: Wrong Rounding of the prices in the percentage discount catalog promotions
**Sylius version affected**: 1.11 and above

**Description**  
Percentage discount in catalog promotions makes prices rounding not correct for edge case prices. I.e. new prices €1.205 is rounded to €1.20 instead of €1.21.    

**Steps to reproduce**  

1. Create product with the price of 10.50. 
2. Create and apply catalog promotion with percentage discount of 15% to the product above
3. As a result the new price will be 8.92, but mathematically correct will be 8.93 (exact discounted price is 8.925)

**Possible Solution**  
Reason is in the calculate method of PercentageDiscountPriceCalculator class. Instead of rounding it just casts discounted price to the integer.

Is it worth of fixing for it or there was some logic behind it?

### PR DESCRIPTION
**Pull request by maxmishyn**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.11                  |
| Bug fix?        | yes                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                      |
| Deprecations?   | no |
| Related tickets | fixes #14522                      |
| License         | MIT                                                          |

### CODE REVIEW

**Comment by CoderMaggie** [comment]
@maxmishyn thank you for opening the fix! ❤️ Would you have some time to add behat scenarios that would cover this?

**Comment by maxmishyn** [comment]
@CoderMaggie Yep. I just added behat scenario covering calculations in the subject of this fix

File: features/promotion/applying_catalog_promotions/applying_catalog_promotions_price_calculation.feature [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        When I view product "Peach Juice"
+        Then I should see the product price "$8.93"
+        And I should see the product original price "$10.50"
+
+    @api @ui
+    Scenario: Discounted price is rounded to an upper last digit
+        When I view product "Mango Juice"
+        Then I should see the product price "$8.45"
+        And I should see the product original price "$9.94"
+
```
Comment (Zales0123) [inline features/promotion/applying_catalog_promotions/applying_catalog_promotions_price_calculation.feature:47]: Extra blank line :dancer:

File: src/Sylius/Bundle/CoreBundle/CatalogPromotion/Calculator/PercentageDiscountPriceCalculator.php [hunk 2]
```diff
@@ -27,7 +27,7 @@ public function supports(CatalogPromotionActionInterface $action): bool
 
     public function calculate(ChannelPricingInterface $channelPricing, CatalogPromotionActionInterface $action): int
     {
-        $price = (int) ($channelPricing->getPrice() - ($channelPricing->getPrice() * $action->getConfiguration()['amount']));
+        $price = (int) round($channelPricing->getPrice() - ($channelPricing->getPrice() * $action->getConfiguration()['amount']));
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/CatalogPromotion/Calculator/PercentageDiscountPriceCalculator.php:30]: As it's a change in `PercentageDiscountPriceCalculator` service implementation, we would need to improve its spec as well 🖖

File: features/promotion/applying_catalog_promotions/applying_catalog_promotions_price_calculation.feature [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    In order to check prices calculation
+    As a Visitor
+    I want to see discounted products in the catalog
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store classifies its products as "Soft Drinks"
+        And the store has a "Orange Juice" configurable product
+        And this product belongs to "Soft Drinks"
+        And this product has "Orange Juice" variant priced at "$20"
```
Comment (Zales0123) [inline features/promotion/applying_catalog_promotions/applying_catalog_promotions_price_calculation.feature:12]:
```php
And the store has an "Orange Juice" product priced at "$20"
And this product belongs to "Soft Drinks"
```

would be simpler and has all the required information 🖖 The same with other products

**Review by Zales0123 — COMMENTED** [review]
Thank you Max, for your contribution! 🚀 Small fixes proposed and it would be good to go 🫡

**Comment by GSadee** [comment]
Closing in favour of https://github.com/Sylius/Sylius/pull/14669

**Pull request closed without merging**